components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/listener/OAuthCacheRemoveListenerTest.java

```diff
@@ -0,0 +1,134 @@
+
+    @Mock
+    OAuthServerConfiguration oauthServerConfigurationMock;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        initMocks(this);
+        mockStatic(IdentityUtil.class);
+        mockStatic(OAuthServerConfiguration.class);
+        mockStatic(OAuthCache.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oauthServerConfigurationMock);
+        when(oauthServerConfigurationMock.getTimeStampSkewInSeconds()).thenReturn(3600L);
+    }
+
+    @DataProvider(name = "provideParams")
+    public Object[][] providePostParams() {
+        Cache cache = mock(Cache.class);
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        authenticatedUser.setUserName("USER_NAME");
+        String[] scope = {"PROFILE", "OPEN_ID"};
+        Timestamp issuedTime = new Timestamp(Calendar.getInstance().getTimeInMillis());
+        Timestamp refreshTime = new Timestamp(3600);
+        long validity = 10000;
+        long refreshTokenVaidity = 10500;
+        final AccessTokenDO tokenDO = new AccessTokenDO("CONSUMER_KEY", authenticatedUser, scope,
+                issuedTime, refreshTime, validity, refreshTokenVaidity, "CODE");
+
+        CacheEntryEvent<? extends OAuthCacheKey, ? extends CacheEntry> cacheEntryEventQualified =
+                new CacheEntryEvent<OAuthCacheKey, CacheEntry>(cache) {
+            OAuthCacheKey cacheKey = new OAuthCacheKey("CACHE_KEY");
+
+            CacheEntry cacheEntry = tokenDO;
+
+            @Override
+            public OAuthCacheKey getKey() {
+                return cacheKey;
+            }
+
+            @Override
+            public CacheEntry getValue() {
+                return cacheEntry;
+            }
+        };
+
+        CacheEntryEvent<? extends OAuthCacheKey, ? extends CacheEntry> cacheEntryEventNullInstance =
+                new CacheEntryEvent<OAuthCacheKey, CacheEntry>(cache) {
+            OAuthCacheKey cacheKey = new OAuthCacheKey("CACHE_KEY");
+
+            @Override
+            public OAuthCacheKey getKey() {
+                return cacheKey;
+            }
+
+            @Override
+            public CacheEntry getValue() {
+                return null;
+            }
+        };
+
+        return new Object[][]{
+                {cacheEntryEventNullInstance}, {cacheEntryEventQualified}
+        };
+    }
+
+    @Test(dataProvider = "provideParams")
+    public void testEntryRemoved(Object object) throws Exception {
+        OAuthCache mockedOAuthCache = mock(OAuthCache.class);
+        when(IdentityUtil.isUserStoreInUsernameCaseSensitive(anyString())).thenReturn(true);
+        when(OAuthCache.getInstance()).thenReturn(mockedOAuthCache);
+        OAuthCacheRemoveListener listener = new OAuthCacheRemoveListener();
+        listener.entryRemoved((CacheEntryEvent<? extends OAuthCacheKey, ? extends CacheEntry>) object);
+
+        when(IdentityUtil.isUserStoreInUsernameCaseSensitive(anyString())).thenReturn(false);
```

Comment:
Better to add this parameter to the data provider as well rather than defining it as two cases in the same method.